### PR TITLE
media/voice: Get I2S card id and device id

### PR DESCRIPTION
**Pull request by junmin-kim**
Use I2S after EPD

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [19f8b5c354e472c202526e9e01a79e0ad19cd0cd] - Code Rule Check Result:
framework/src/media/voice/HardwareEndPointDetector.h:28: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/src/media/voice/HardwareKeywordDetector.h:28: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)

**Comment by seinfra** [comment]
Target : [19f8b5c354e472c202526e9e01a79e0ad19cd0cd] - Code Rule Check (C++) OK.

**Comment by junmin-kim** [comment]
Now I'm testing it

File: framework/src/media/voice/HardwareEndPointDetector.h [hunk 1]
```diff
@@ -28,16 +28,18 @@ namespace voice {
...
 
 public:
 	bool init(uint32_t samprate, uint8_t channels) override;
 	void deinit() override;
 	bool startEndPointDetect(int timeout) override;
 
 private:
-	int mCard;
-	int mDevice;
+	int mNormalCard;
```
Comment (Taejun-Kwon) [inline framework/src/media/voice/HardwareEndPointDetector.h]: Could you please add description regarding these variables??

File: framework/src/media/voice/SpeechDetector.cpp [hunk 2]
```diff
@@ -61,24 +61,23 @@ bool SpeechDetectorImpl::initKeywordDetect(uint32_t samprate, uint8_t channels)
 		return false;
 	}
 
-	int card = -1;
-	int device = -1;
+	int sd_card = -1;
```
Comment (Taejun-Kwon) [inline framework/src/media/voice/SpeechDetector.cpp]: Let's define INVALID_HW_NUMBER or NONE_HW_NUMBER something else. 
To improve code understanding.

File: framework/src/media/voice/SpeechDetector.cpp [hunk 3]
```diff
@@ -61,24 +61,23 @@ bool SpeechDetectorImpl::initKeywordDetect(uint32_t samprate, uint8_t channels)
...
-		medvdbg("KeywordDetector : card_id : %d device_id %d\n", card, device);
-		mKeywordDetector = std::make_shared<HardwareKeywordDetector>(card, device);
-
-		result = change_stream_in_device(card, device);
-		if (result != AUDIO_MANAGER_SUCCESS && result != AUDIO_MANAGER_DEVICE_ALREADY_IN_USE) {
-			meddbg("change_stream_in_device failed: %d\n", result);
-			mKeywordDetector = nullptr;
-			return false;
-		}
+		medvdbg("KeywordDetector : card_id : %d device_id %d\n", sd_card, sd_device);
```
Comment (Taejun-Kwon) [inline framework/src/media/voice/SpeechDetector.cpp]: Please add description about this. 'why this is needed?'

File: framework/src/media/voice/SpeechDetector.cpp (hunk 2 above)
Comment (junmin-kim) [reply framework/src/media/voice/SpeechDetector.cpp]: I'll replace with INVALID_HW_NUMBER

**Comment by seinfra** [comment]
Target : [420e5f7b7c2416920818d476619b82de9b1a1d8c] - Code Rule Check Result:
framework/src/media/voice/HardwareEndPointDetector.h:28: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/src/media/voice/HardwareKeywordDetector.h:28: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)

**Comment by seinfra** [comment]
Target : [420e5f7b7c2416920818d476619b82de9b1a1d8c] - Code Rule Check (C++) OK.

**Comment by junmin-kim** [comment]
@Taejun-Kwon 
Currently we find card id at init time.
What if card id is not available at deinit time?

I think we should find and change device at same time.
How do you think?

**Comment by seinfra** [comment]
Target : [e747863a8d0ce48fd5fc4b44e424af2141975d59] - Code Rule Check Result:
framework/src/media/voice/HardwareEndPointDetector.h:28: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/src/media/voice/HardwareKeywordDetector.h:28: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)

**Comment by seinfra** [comment]
Target : [e747863a8d0ce48fd5fc4b44e424af2141975d59] - Code Rule Check (C++) OK.

File: framework/src/media/voice/HardwareEndPointDetector.h (hunk 1 above)
Comment (junmin-kim) [reply framework/src/media/voice/HardwareEndPointDetector.h]: I added comments

File: framework/src/media/voice/SpeechDetector.cpp (hunk 3 above)
Comment (junmin-kim) [reply framework/src/media/voice/SpeechDetector.cpp]: finding AUDIO_DEVICE_PROCESS_TYPE_NONE will be implemented later

**Comment by seinfra** [comment]
Target : [0c9d60aeb442a36614c7b96f314df4f5aa39cade] - Code Rule Check Result:
framework/src/media/voice/HardwareEndPointDetector.h:28: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/src/media/voice/HardwareKeywordDetector.h:28: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)

**Comment by seinfra** [comment]
Target : [0c9d60aeb442a36614c7b96f314df4f5aa39cade] - Code Rule Check (C++) OK.

**Pull request merged**